File: js/2660choosePresident.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let memberCount = 0;
let friendRelationships = [];

rl.on("line", function (line) {
  const [first, second] = line.split(" ").map(Number);

  if (first === -1 && second === -1) {
    rl.close();
  }
  if (memberCount === 0) {
    memberCount = first;
    friendRelationships = Array.from({ length: memberCount + 1 }, () =>
      Array(memberCount + 1).fill(null)
    );
  } else {
    friendRelationships[first][second] = 1;
    friendRelationships[second][first] = 1;
  }
});

rl.on("close", function () {
  console.log(friendRelationships);

  // i, j != 0 , i!==j
  for (let i = 1; i < memberCount + 1; i++) {
    let thisLine = [];
    for (let j = 1; j < i; j++) {
      
    }
  }
});

File: js/2660choosePresident.py
memberCount = int(input())

friendRelationships = [[None for _ in range(memberCount+1)] for _ in range(memberCount+1)]

while(True):
    a, b = map(int, input().split(' '))
    
    if(a==-1 and b==-1):
      break
    friendRelationships[a][b] = 1
    friendRelationships[b][a] = 1

def getFriendList(member):
   return friendRelationships[member][1:member]

def getVerticalFriendList(member):
   friendList = []
   for i in range(member+1, memberCount+1):
      friendList.append(friendRelationships[i][member])
   return friendList

for member in range(1, memberCount+1):
   friendList = getFriendList(member)
   for j in range(1, len(friendList)):
      if(friendList[j]!=None):
        friendList2 = getVerticalFriendList(j)
        bestChoice = memberCount+1
        print('for friend', j, ':', friendList2)
        for k in range(1, len(friendList2)):
           if(friendList2[k] and friendList[friendList2[k]] and friendList2[k]+1 < bestChoice):
              bestChoice= friendList2[k]+1
        friendRelationships[member][j] = None if(bestChoice == memberCount+1) else bestChoice

# for i in friendRelationships:
#    print(i)         
